plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2058,10 +2058,38 @@ public void testCreateTableUnsupportedPartitionTypeAs()
                 .hasMessageMatching("Unsupported type .* for partition: a");
     }
 
-    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "Unsupported Hive type: varchar\\(65536\\)\\. Supported VARCHAR types: VARCHAR\\(<=65535\\), VARCHAR\\.")
+    @Test(expectedExceptions = RuntimeException.class)
     public void testCreateTableNonSupportedVarcharColumn()
     {
-        assertUpdate("CREATE TABLE test_create_table_non_supported_varchar_column (apple varchar(65536))");
+        assertUpdate("CREATE TABLE test_create_table_non_supported_varchar_column (apple varchar(" + ((long) Integer.MAX_VALUE + 1) + "))");
+    }
+
+    @Test
+    public void testCreateTableSupportedVarcharColumns()
+    {
+        assertUpdate("CREATE TABLE test_create_table_supported_varchar_columns_apple (apple varchar(65534))");
+        TableMetadata tableMetadataApple = getTableMetadata(catalog, TPCH_SCHEMA, "test_create_table_supported_varchar_columns_apple");
+        assertColumnType(tableMetadataApple, "apple", createVarcharType(65534));
+
+        assertUpdate("CREATE TABLE test_create_table_supported_varchar_columns_banana (banana varchar(65535))");
+        TableMetadata tableMetadataBanana = getTableMetadata(catalog, TPCH_SCHEMA, "test_create_table_supported_varchar_columns_banana");
+        assertColumnType(tableMetadataBanana, "banana", createVarcharType(65535));
+
+        assertUpdate("CREATE TABLE test_create_table_supported_varchar_columns_cherry (cherry varchar(65536))");
+        TableMetadata tableMetadataCherry = getTableMetadata(catalog, TPCH_SCHEMA, "test_create_table_supported_varchar_columns_cherry");
+        assertColumnType(tableMetadataCherry, "cherry", createUnboundedVarcharType());
```

Comment:
Isn't this a bit misleading like we create a table which has a limit constraint of length as `65536` but now we are implicitly casting to `UnboundedVarchar`

Reply:
@Praveen2112 , This enhancement is focusing on translating Trino SPI VARCHAR of length greater than 65535 to Hive STRING as that is the appropriate data type that can fit the data in Hive without any loss. This type of automatic type translation/coercion is done elsewhere in Trino not just with character/string types but also with dates/timestamps.